services/motion/builtin/move_request.go

```diff
@@ -837,3 +842,28 @@ func toGeoPosePlanSteps(posesByComponent []motionplan.PlanStep, geoPoses []spati
 	}
 	return steps, nil
 }
+
+// This function will check the orientation of the startPose of a 2d move request, and ensure that it is normal to the XY plane. If it is
+// not, it will be altered such that it is (accounting for e.g. an ourdoor base with one wheel on a rock). If the orientation is such that
+// the robot is pointed directly up or down (or is upside-down), an error is returned.
+func correctStartPose(startPose spatialmath.Pose) (spatialmath.Pose, error) {
+	orient := startPose.Orientation().OrientationVectorRadians() // OV is easy to check if we are in plane
+	if orient.OZ < 0 {
+		return nil, errors.New("base appears to be upside down, check your movement sensor")
+	}
+	if orient.OX == 0 && orient.OY == 0 {
+		return startPose, nil
+	}
+
+	adjPt := spatialmath.NewPoseFromPoint(r3.Vector{0, 1, 0})
+	newAdjPt := spatialmath.Compose(spatialmath.NewPoseFromOrientation(orient), adjPt).Point()
```

Comment:
Can you comment this function more?  Its a little hard to get an intuition for what you are doing

Reply:
Done